src/main/java/li/cil/tis3d/api/FontRendererAPI.java

```diff
@@ -1,23 +1,26 @@
 package li.cil.tis3d.api;
 
+import net.fabricmc.api.EnvType;
+import net.fabricmc.api.Environment;
+import net.minecraft.client.render.VertexConsumer;
+import net.minecraft.client.render.VertexConsumerProvider;
+import net.minecraft.client.util.math.MatrixStack;
+
 /**
  * API entry point for access to the tiny font renderer used on execution
  * modules, for example.
  * <p>
  * Keep in mind that the list of printable characters is very small for this
  * font renderer, due to the small number of characters in the font sheet.
  */
-public final class FontRendererAPI {
+@Environment(EnvType.CLIENT)
+public interface FontRendererAPI {
     /**
      * Render the specified string.
      *
      * @param string the string to render.
      */
-    public static void drawString(final String string) {
-        if (API.fontRendererAPI != null) {
-            API.fontRendererAPI.drawString(string);
-        }
-    }
+    void drawString(final String string);
```

Comment:
Oh, I think I might have just remembered why I wasn't dog-fooding some of the text rendering; it's because not using the API in the UI classes would allow me to remove these not-VertexConsumer-using legacy GL methods already (the UI classes are their only remaining users).

Reply:
Cool, as mentioned above, let's wait with that then.